### PR TITLE
Remove github.com/otiai10/copy dependency

### PR DESCRIPTION
**Pull request by alexandear**
This PR replaces usage of [`copy.Copy`](https://pkg.go.dev/github.com/otiai10/copy#Copy) with hand-written `copyDir` function. It's not worth keeping this dependency used only in tests.

Yeah, `copyDir` is not ideal, and `copy.Copy` is much more powerful and covers many of OS's corner cases, but for tests it's good. In the future, we can return this dependency if `copy.Copy` will be needed for the prod code.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@mtrmac PTAL

File: oci/layout/oci_delete_test.go [hunk 1]
```diff
@@ -296,3 +295,61 @@ func assertBlobDoesNotExist(t *testing.T, blobsDir string, blobDigest string) {
...
+
+	for _, dirEntry := range dirEntries {
+		srcPath := filepath.Join(src, dirEntry.Name())
+		dstPath := filepath.Join(dst, dirEntry.Name())
+
+		copy := copyFile
+		if dirEntry.IsDir() {
+			copy = copyDir
+		}
+
```
Comment (mtrmac) [inline oci/layout/oci_delete_test.go:323]: Using `if … copyFile` else … `copyDir` here would be simpler than adding the variable.

**Review by mtrmac — COMMENTED** [review]
Thanks!

I don’t feel strongly about the change _itself_.

It’s just… This is a unit test. It does not show up in any dependency of c/image. So why is it important to trim dependencies? To me, adding and reviewing extra ~55 lines of code for no end-user benefit seems like a bad trade. So what am I missing? Are there any negative effects of that dependency I don’t know about, which would also apply to _other_ dependencies in the future?

**Comment by alexandear** [comment]
@mtrmac The [README](https://github.com/containers/image/blob/main/README.md) indeed categorizes `containers/image` as "image is a set of Go libraries", implying its main purpose is to be imported by users. Each time a user imports the package, it automatically imports its dependencies. This reduces download speed and increases the user's application dependency list. To maintain efficient use and management, it's critical to keep a minimal list of dependencies. 

Adding to this, having fewer dependencies enhances the overall control we have on the application. Reviewing and adding 55 lines of code is a one-time task. But every extra dependency is an added layer of complexity, potential security risk, and another component that should be maintained and kept updated. We are simplifying the application maintenance process by trimming down dependencies, just as Rob Pike mentioned about keeping the dependency tree small (check [video](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=598s)).

https://medium.com/@cep21/aspects-of-a-good-go-library-7082beabb403

File: oci/layout/oci_delete_test.go (hunk 1 above)
Comment (alexandear) [reply oci/layout/oci_delete_test.go]: done

**Comment by mtrmac** [comment]
> Each time a user imports the package, it automatically imports its dependencies. 

I mean…

```console
% cat foo.go 
package foo

import "github.com/containers/image/v5/oci/layout"

func main() {
  _ = layout.Transport
}
% go mod tidy && go mod vendor
% grep -lr 10/copy
./go.sum
```
Test-only dependencies don’t really “infect” the consumer.

**Review by mtrmac — COMMENTED** [review]
The recent commits need to follow https://github.com/containers/image/blob/main/CONTRIBUTING.md#sign-your-prs .

(And eventually, please squash them into one.)

**Comment by alexandear** [comment]
Actually, you're right. Test dependencies won't inflect customer since Go 1.18. Found this change
https://go-review.googlesource.com/c/go/+/357310/

Closing PR.
